evalml/pipelines/component_graph.py

```diff
@@ -83,89 +88,95 @@ def fit(self, X, y):
         """Fit each component in the graph
 
         Arguments:
-            X (pd.DataFrame): The input training data of shape [n_samples, n_features]
-            y (pd.Series): The target training data of length [n_samples]
+            X (ww.DataTable, pd.DataFrame): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series): The target training data of length [n_samples]
         """
         self._compute_features(self.compute_order, X, y, fit=True)
         return self
 
     def fit_features(self, X, y):
-        """ Fit all components save the final one, usually an estimator
```

Comment:
There was a lot of duplicate code in `fit_features` and` compute_final_component_features` so I refactored using a new helper function, `_fit_transform_features_helper`